hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -307,6 +308,143 @@ restartedNodeCanAbort tracer workDir cardanoNode hydraScriptsTxId = do
+          , postDecommit n1 decommitTx
+          ]
+
+      waitFor hydraTracer 10 [n1, n2] $
+        output "DecommitRequested" ["headId" .= headId, "decommitTx" .= decommitTx, "utxoToDecommit" .= decommitUTxO]
+      waitFor hydraTracer 10 [n1, n2] $
+        output "DecommitApproved" ["headId" .= headId, "decommitTxId" .= decommitTxId, "utxoToDecommit" .= decommitUTxO]
+      failAfter 10 $ waitForUTxO cardanoNode decommitUTxO
+
+      distributedUTxO <- waitForAllMatch 10 [n1, n2] $ \v -> do
+        guard $ v ^? key "tag" == Just "DecommitFinalized"
+        guard $ v ^? key "headId" == Just (toJSON headId)
+        v ^? key "distributedUTxO" . _JSON
+
+      guard $ distributedUTxO `UTxO.containsOutputs` utxoFromTx decommitTx
+
+      pure (headId, decommitUTxO)
+
+    -- Here we post a deposit while one node is down so we can test if recover works later on
+    resp2 <-
+      parseUrlThrow ("POST " <> hydraNodeBaseUrl n1 <> "/commit")
+        <&> setRequestBodyJSON commitUTxO2
+          >>= httpJSON
+
+    let depositTransaction2 = getResponseBody resp2 :: Tx
+    let tx2 = signTx bobCardanoSk depositTransaction2
+
+    submitTx cardanoNode tx2
+
+    threadDelay $ fromIntegral (deadline * 2 + 1)
+
+    bobChainConfigFromTip <-
+      chainConfigFor Bob workDir nodeSocket hydraScriptsTxId [Alice] contestationPeriod depositDeadline
+        <&> modifyConfig (\config -> config{networkId, startChainFrom = Just tip})
+
+    withTempDir "blank-state" $ \tmpDir -> do
+      void $ readCreateProcessWithExitCode (proc "cp" ["-r", workDir </> "state-2", tmpDir]) ""
+      void $ readCreateProcessWithExitCode (proc "rm" ["-rf", tmpDir </> "state-2" </> "state"]) ""
+      withHydraNode hydraTracer bobChainConfigFromTip tmpDir 2 bobSk [aliceVk] [1] $ \n2 -> do
+        -- Also expect to see past server outputs replayed
+        headId2 <- waitMatch 5 n2 $ headIsInitializingWith (Set.fromList [alice, bob])
+        headId2 `shouldBe` headId1
+        waitFor hydraTracer 5 [n2] $
+          output "HeadIsOpen" ["utxo" .= object mempty, "headId" .= headId2]
+
+        distributedUTxO <- waitForAllMatch 5 [n2] $ \v -> do
+          guard $ v ^? key "tag" == Just "DecommitFinalized"
+          guard $ v ^? key "headId" == Just (toJSON headId2)
+          v ^? key "distributedUTxO" . _JSON
+
+        guard $ distributedUTxO `UTxO.containsOutputs` decrementOuts
+
+        let depositTxId = getTxId $ getTxBody depositTransaction2
+        let path = BSC.unpack $ urlEncode False $ encodeUtf8 $ T.pack $ show depositTxId
+
+        recoverResp <-
+          parseUrlThrow ("DELETE " <> hydraNodeBaseUrl n2 <> "/commits/" <> path)
+            >>= httpJSON
+
+        (getResponseBody recoverResp :: String) `shouldBe` "OK"
+
+        waitForAllMatch (20 * blockTime) [n2] $ \v -> do
+          guard $ v ^? key "tag" == Just "CommitRecovered"
+          guard $ v ^? key "recoveredTxId" == Just (toJSON depositTxId)
```

Comment:
minor: i think it might be cool to continue the lifecycle and see how close/fanout look, wdyt?